src/services/utils.ts

```diff
@@ -12,13 +12,47 @@ export const backendFetch = (url: string, init: any, token?: string) => {
     return safeFetch(url, initCopy);
 };
 
-export const backendFetchJson = (url: string, init: any, token?: string) => {
-    const initCopy = prepareRequest(init, token);
-    return safeFetch(url, initCopy).then((safeResponse) =>
+export const backendFetchJson = (url: string, init?: any, token?: string) => {
+    return backendFetch(url, init, token).then((safeResponse) =>
         safeResponse.status === 204 ? null : safeResponse.json()
     );
 };
 
+export const backendFetchText = (url: string, init?: any, token?: string) => {
+    return backendFetch(url, init, token).then((safeResponse) =>
+        safeResponse.text()
+    );
+};
+
+export const backendFetchFile = (url: string, init?: any, token?: string) => {
+    return backendFetch(url, init, token).then((safeResponse) =>
+        safeResponse.blob()
+    );
+};
```

Comment:
`backendFetchText`, `backendFetchFile` `backendFetch` were accepting all 200 range return codes, won't restricting it to only 204 be prone to issues ?

Reply:
I would let as it is personnaly

Reply:
Normally no, because a response 204 specifically say that there is no content in the response.  
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses  
> [204 No Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204)
    There is no content to send for this request, but the headers may be useful. The user agent may update its cached headers for this resource with the new ones.

Reply:
I just noticed gridexplore's version didn't have the condition on 204, so I guess it's alright